test-common/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+use substrate_archive::{database::models::BlockModel, native_executor_instance};
+use anyhow::Error;
+use paste::paste;
+use polkadot_service::{polkadot_runtime as dot_rt, Block};
+use std::{fs::File, sync::Arc};
+use substrate_archive_backend::{
+	runtime_api, ExecutionMethod, ReadOnlyBackend, ReadOnlyDb, RuntimeConfig, SecondaryRocksDb, TArchiveClient,
+};
+
+native_executor_instance!(
+	pub PolkadotExecutor,
+	dot_rt::api::dispatch,
+	dot_rt::native_version,
+	sp_io::SubstrateHostFunctions,
+);
+
+type PolkadotClient = TArchiveClient<Block, dot_rt::RuntimeApi, PolkadotExecutor, SecondaryRocksDb>;
+type PolkadotBackend = Arc<ReadOnlyBackend<Block, SecondaryRocksDb>>;
+
+pub fn get_dot_runtime_api(
+	block_workers: usize,
+	wasm_pages: u64,
+) -> Result<(PolkadotClient, PolkadotBackend), Error> {
+	let chain_path = std::env::var("CHAIN_DATA_DB").expect("must have database as variable CHAIN_DATA_DB");
+	let mut tmp_path = tempfile::tempdir()?.into_path();
+	tmp_path.push("rocksdb_secondary");
+	let db = Arc::new(SecondaryRocksDb::open_database(chain_path.as_str(), 128, tmp_path).unwrap());
```

Comment:
This is actually a config parameter configurable via the `.toml`, but since this is a test it's just hardcoded. However, even in production I haven't found much benefit to increasing beyond 128